### PR TITLE
fixes and improvements for the WebNLG loader

### PR DESCRIPTION
**Pull request by Shimorina**
- fixes test sets loading in v3.0
- adds additional fields for v3.0_ru
- adds info to the WebNLG data card

### CODE REVIEW

**Comment by yjernite** [comment]
The dataset card is fantastic!

Looks good to me! Did you check that this still passes the slow tests with the existing dummy data?

**Comment by Shimorina** [comment]
Yes, I ran and passed all the tests specified in [this guide](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#automatically-add-code-metadata), including the slow ones.

File: datasets/web_nlg/README.md [hunk 1]
```diff
@@ -157,70 +157,110 @@ A typical example contains the original RDF triples in the set, a modified versi
...
 - `lex`: the lexicalizations, with:
   - `text`: the text to be predicted.
-  - `lid`: a lexicalizayion ID, unique per example.
+  - `lid`: a lexicalization ID, unique per example.
   - `comment`: the lexicalizations were rated by crowd workers are either `good` or `bad`
+  - `lang`: (for `release_v3.0_ru`) the language used because original English texts were kept in the Russian version.
+
+Russian data has additional optional fields comparing to English:
+- `dbpedialinks`: RDF triples extracted from DBpedia between English and Russian entities by means of the property `sameAs`.
+- `links>`: RDF triples created manually for some entities to serve as pointers to translators. There are two types of them:
```
Comment (lhoestq) [inline datasets/web_nlg/README.md:174]:
```suggestion
- `links`: RDF triples created manually for some entities to serve as pointers to translators. There are two types of them:
```
small typo

File: datasets/web_nlg/web_nlg.py [hunk 2]
```diff
@@ -238,8 +252,15 @@ def _generate_examples(self, filedirs):
         id_ = 0
         for xml_location in filedirs:
             for xml_file in sorted(glob(pjoin(xml_location, "*.xml"))):
-                test_cat = xml_file.split("/")[-1][:-4] if "webnlg_challenge_2017/test" in xml_file else ""
+                if (
+                    "webnlg_challenge_2017/test" in xml_file
+                    or "release_v3.0/en/test" in xml_file
+                    or "release_v3.0/ru/test" in xml_file
+                ):
+                    test_cat = xml_file.split("/")[-1][:-4]
```
Comment (lhoestq) [inline datasets/web_nlg/web_nlg.py:260]: Unfortunately this kind of check may not work on Windows. Indeed Windows uses backslashes and not simple slashes between directory names.

So I would change it this way:
```suggestion
                # windows may use backslashes so we first need to replace them with slashes
                xml_file_path_with_slashes = "/".join(Path(xml_file).parts)
                if (
                    "webnlg_challenge_2017/test" in xml_file_path_with_slashes
                    or "release_v3.0/en/test" in xml_file_path_with_slashes
                    or "release_v3.0/ru/test" in xml_file_path_with_slashes
                ):
                    test_cat = xml_file_path_with_slashes.split("/")[-1][:-4]
```
Let me know what you think

**Review by lhoestq — COMMENTED** [review]
Thanks for the changes ! And good dataset card as well :)

Looks all good to me, I just left a comment for a typo and a comment about paths manipulations that may not work on windows

**Comment by lhoestq** [comment]
I just added the `from pathlib import Path` at the top to fix the script

**Review by lhoestq — APPROVED** [review]
Looks all good now :)
Thanks you for the changes !

**Comment by lhoestq** [comment]
I ran the tests locally and they all pass, merging

**Comment by Shimorina** [comment]
Thank you for the review!

**Pull request merged**